src/main/scala/com/databricks/spark/xml/XmlReader.scala

```diff
@@ -112,6 +117,23 @@ class XmlReader extends Serializable {
     spark.baseRelationToDataFrame(relation)
   }
 
+  /**
+   * @param spark current SparkSession
+   * @param df XML for individual 'rows' as Strings
+   * @return XML parsed as a DataFrame
+   */
+  def xmlDataFrame(spark: SparkSession, df: DataFrame): DataFrame = {
```

Comment:
Oh, wait, how about changing `DataFrame` -> `Dataset[String]`?

Reply:
I figured the common case might be parsing the result of `myDF.select("someXML")`.
Want to add xmlDataset as well for Dataset[String]? it's trivial.

Reply:
I just wanted to make it matched to existing Spark API like https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala#L506

Actually I have an idea about adding an implicit too so that our library allows `spark.read.xml(dataset)`. Let me push some changes into this branch.

Reply:
Actually, let me just merge and make a separate PR. Can you resolve conflicts?

Reply:
Yeah go for it, any helper or convenience methods here are probably worthwhile.